srtcore/core.cpp

```diff
@@ -8017,30 +8041,25 @@ void srt::CUDT::sendCtrl(UDTMessageType pkttype, const int32_t* lparam, void* rp
         m_tsLastSndTime.store(steady_clock::now());
 }
 
-// [[using locked(m_RcvBufferLock)]]
 bool srt::CUDT::getFirstNoncontSequence(int32_t& w_seq, string& w_log_reason)
 {
+    if (!m_pRcvBuffer)
     {
-        ScopedLock losslock (m_RcvLossLock);
-        const int32_t seq = m_pRcvLossList->getFirstLostSeq();
-        if (seq != SRT_SEQNO_NONE)
-        {
-            HLOGC(xtlog.Debug, log << "NONCONT-SEQUENCE: first loss %" << seq << " (loss len=" <<
-                    m_pRcvLossList->getLossLength() << ")");
-            w_seq = seq;
+        LOGP(cnlog.Error, "IPE: ack can't be sent, buffer doesn't exist and no group membership");
+        return false;
+    }
+    {
```

Comment:
Is there a need for this extra block? It is followed by the return statement.

Reply:
Things changed after removed code and something that looks like a past-TSAN fix. Yes, should be dissolved.